python/ray/experimental/channel/nccl_group.py

```diff
@@ -277,7 +277,70 @@ def allreduce(
         # need to synchronize here and check that the channel is still open to
         # ensure that the receive buffer is valid.
         # TODO(swang): Avoid CUDA synchronization.
-        # TODO(wxdeng): Use check_async_error.
+        # This synchronize will be optional after merging the unify PR.
+        self._cuda_stream.synchronize()
+        if self._closed:
+            raise RayChannelError(
+                "NCCL group has been destroyed during allreduce operation. "
+                "There may be a dtype mismatch between input tensors from "
+                "different ranks."
+            )
+
+    def reducescatter(
+        self,
+        send_buf: "torch.Tensor",
+        recv_buf: "torch.Tensor",
+        op: ReduceScatterOp = ReduceScatterOp.SUM,
+    ):
+        if self._closed:
```

Comment:
Please avoid code duplication and separate out the common code across these methods into a helper function.